### PR TITLE
Problem: v1.0.x changes are not merged into main

### PR DESCRIPTION
**Pull request by yihuang**
👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,112 @@
 # Changelog
 
-## Unreleased
+*Jan 19, 2023*
+
+## v1.0.3
```
Comment (yihuang) [inline CHANGELOG.md:5]:
```suggestion
## Unreleased
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,112 @@
...
+
+## v0.9.0-beta3
+
+### Bug Fixes
+
+- [cronos#696](https://github.com/crypto-org-chain/cronos/pull/696) Fix json-rpc apis for legacy blocks.
+
+*Aug 29, 2022*
+
+## v0.9.0-beta2
```
Comment (thomas-nguy) [inline CHANGELOG.md:107]: hmm I think some of the PR (especially those related to gravity like the turnbridge #775) were not part of the august release but needs to be move to unreleased

Comment (yihuang) [reply CHANGELOG.md:116]: updated, is it correct now?

Comment (thomas-nguy) [reply CHANGELOG.md:116]: yes I think so

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,13 +3,118 @@
...
+
 ### Improvements
 
 - [cronos#702](https://github.com/crypto-org-chain/cronos/pull/702) Integrate the file state streamer.
```
Comment (mmsqe) [inline CHANGELOG.md:120]: seems [v0.9.0-beta2](https://github.com/crypto-org-chain/cronos/blob/v0.9.0-beta2/cmd/cronosd/cmd/root.go) dont have change like [FlagStreamers](https://github.com/crypto-org-chain/cronos/pull/721/files) or [SetIAVLDisableFastNode](https://github.com/crypto-org-chain/cronos/pull/714/files), we include in [v0.9.0-beta4](https://github.com/crypto-org-chain/cronos/blob/v0.9.0-beta4/cmd/cronosd/cmd/root.go)

File: integration_tests/test_upgrade.py [hunk 4]
```diff
@@ -163,13 +171,16 @@ def test_cosmovisor_upgrade(custom_cronos: Cronos):
...
+    assert old_erc20_balance == contract.caller(block_identifier=old_height).balanceOf(
+        ADDRS["validator"]
+    )
+
+    assert not cli.evm_params()["params"]["extra_eips"]
+
+    # check the gas cost is lower after upgrade
+    assert (
+        old_gas - 3700
+        == contract.functions.transfer(ADDRS["community"], 100).buildTransaction(
```
Comment (mmsqe) [inline integration_tests/test_upgrade.py:183]:
```suggestion
        == contract.functions.transfer(ADDRS["community"], 100).build_transaction(
```

File: integration_tests/test_upgrade.py [hunk 5]
```diff
@@ -98,9 +98,17 @@ def test_cosmovisor_upgrade(custom_cronos: Cronos):
...
     old_balance = w3.eth.get_balance(ADDRS["validator"], block_identifier=old_height)
     old_base_fee = w3.eth.get_block(old_height).baseFeePerGas
+    old_erc20_balance = contract.caller(block_identifier=old_height).balanceOf(
+        ADDRS["validator"]
+    )
     print("old values", old_height, old_balance, old_base_fee)
 
-    plan_name = "v0.9.0"
+    # estimateGas for an erc20 transfer tx
+    old_gas = contract.functions.transfer(ADDRS["community"], 100).buildTransaction(
```
Comment (mmsqe) [inline integration_tests/test_upgrade.py:107]:
```suggestion
    old_gas = contract.functions.transfer(ADDRS["community"], 100).build_transaction(
```

**Comment by tomtau** [comment]
I guess for the next time, it'll be better to first patch on main and then backport to release branches: https://trunkbaseddevelopment.com/branch-for-release/ ?

**Comment by yihuang** [comment]
> I guess for the next time, it'll be better to first patch on main and then backport to release branches: https://trunkbaseddevelopment.com/branch-for-release/ ?

For many changes, they need to be tested on testnet/mainnet first, but main branch is consensus breaking, thus current workflow.

**Review by thomas-nguy — APPROVED** [review]

**Comment by thomas-nguy** [comment]
Some changes such as dependencies upgrades and bugfix could be tested and done on main first and cherry picked on release branches

for patches specific to release branch, some may not even need to be rebased on main

File: integration_tests/configs/default.jsonnet [hunk 6]
```diff
@@ -2,7 +2,7 @@
   dotenv: '../../scripts/.env',
   'cronos_777-1': {
     cmd: 'cronosd',
-    'start-flags': '--trace --streamers file',
+    'start-flags': '--trace file',
```
Comment (yihuang) [inline integration_tests/configs/default.jsonnet:5]:
```suggestion
    'start-flags': '--trace',
```

**Review by mmsqe — APPROVED** [review]

**Pull request merged**